spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/context/RoundRobinLoadBalancer.java

```diff
@@ -0,0 +1,90 @@
+package org.springframework.cloud.client.loadbalancer.impl.context;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.impl.scoped.ScopedDiscoveryClient;
+import org.springframework.cloud.client.loadbalancer.impl.support.LoadBalancerClientFactory;
+
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * @author Spencer Gibb
+ */
+public class RoundRobinLoadBalancer implements LoadBalancer<ServiceInstance> {
```

Comment:
Indeed. In fact in some of our new work involving gRPC we enforce a consistent sorting on the servers coming back from eureka. Debating now on if/how we should roll that out into the OSS eureka server.